implementation/operations.go

```diff
@@ -0,0 +1,144 @@
+package implementation
+
+import (
+	"errors"
+	"github.com/aditya/ProjectCatalog/models"
+	"sort"
+	"strconv"
+	"time"
+)
+
+type Inmemoryimplement struct{
+	ProductArr []models.Product
```

Comment:
this should be a private member of the class.

Reply:
name can be `products`, also should we use map instead of slice, searching by id would be inefficient in the slice.